flows/actions/send_msg.go

```diff
@@ -155,26 +157,51 @@ func (a *SendMsgAction) getTemplateMsg(run flows.Run, urn urns.URN, channelRef *
 		evaluatedParams[comp.Name] = evaluatedCompParams
 	}
 
-	// next we cross reference with params defined in the template translation to get types
-	params := make(map[string][]flows.TemplateParam, len(translation.Components()))
+	oldParams := make(map[string][]flows.TemplateParam, len(translation.Components()))
 
 	for key, comp := range translation.Components() {
 		compParams := comp.Params()
 		if len(compParams) > 0 {
-			params[key] = make([]flows.TemplateParam, len(compParams))
+			oldParams[key] = make([]flows.TemplateParam, len(compParams))
+		}
+
+		for i, tp := range compParams {
+			if i < len(evaluatedParams[key]) {
+				oldParams[key][i] = flows.TemplateParam{Type: tp.Type(), Value: evaluatedParams[key][i]}
+			} else {
+				oldParams[key][i] = flows.TemplateParam{Type: tp.Type(), Value: ""}
+			}
 		}
+	}
+
+	// next we cross reference with params defined in the template translation to get types
+	components := make([]flows.TemplateComponent, 0)
+
+	translationComponents := translation.Components()
+
+	compKeys := maps.Keys(translationComponents)
+	sort.Strings(compKeys)
```

Comment:
I think we want to...

1. Return components in the same order we're given them
2. Return the same number of components as we're given, even if there are no params

Reply:
I think `translation.Components()` is still the components as a dict so I think we'll use the same order when we use the components as a list

Reply:
Yeah so then this PR needs to include the switch to components as a list...

Reply:
ok

Reply:
And I'll hold off making any mailroom changes